### DISCUSSION TITLE
Filter date with interval selector in a mark_geoshape (ALTAIR)

### DISCUSSION THREAD

**Discussion by arnaupo** [vega/altair#3716] [Q&A]
I want to do a choropleth map that you can filter the date range with an interval selector (It has to be an interval selector because it also changes other charts in the dashboard). I do the interval selection on the following chart:

![Captura de pantalla 2024-12-20 113634](https://github.com/user-attachments/assets/4c004648-e881-4bc6-9a21-8a06f8ee2292)

This works well for me with filtering dates in line charts, as I can change the domain directly in the axis. However, in the mark_geoshape I do not have that option.

```
counties = alt.Chart(county_data).transform_lookup(
    lookup='CountyFIPS',
    from_=alt.LookupData(gdf_us_counties, 'id', ['type', 'geometry'])
).mark_geoshape(
    stroke='black')
....
).project(
    type='albersUsa'
).transform_aggregate(
    ss = 'sum(Shootings)',
    groupby = ['CountyFIPS', 'geometry', 'type', 'State', 'County']
).encode(
    tooltip = [alt.Tooltip('State:N'), alt.Tooltip('County:N'), alt.Tooltip('ss:Q', title = 'Total Shootings')],
    color = alt.Color('ss:Q', scale = alt.Scale(scheme='viridis')),
).properties(
    width=600,
    height=450,
    title='Shootings per County'
)
```

I have tried accessing the endpoints of the iterval selector following the steps in [https://github.com/vega/altair/issues/1765](url) using transform calculate: However, it didn't work. I tried to acces the value and printing it with mark text. At first, i get the non-defined value and as soon as I used the filter the text dissappeared, so I assume it is not getting any value.

 The data has the following format:

![Captura de pantalla 2024-12-20 114114](https://github.com/user-attachments/assets/13d84acc-c236-4082-a186-90e03eee5148)

What I want to achieve (and I did by manually filering it) is to plot something like this:

![Captura de pantalla 2024-12-20 114644](https://github.com/user-attachments/assets/06bb5841-110c-4caf-a76e-efa56809aa67)

Thanks in advance!

**Comment by joelostblom** [comment]
Could you upload the full code required to reproduce the issue? Ideally making it as minimal as possible by removing aspects that are not required to reproduce what you are describing.
